app/components/OceanInterface/OceanInterface.tsx

```diff
@@ -21,61 +23,78 @@ async function handlePress (txid: string): Promise<void> {
+      return await broadcastTransaction(tx, client, retries + 1)
     }
-    setError(e)
+    throw e
   }
 }
 
 /**
  * @description - Global component to be used for async calls, network errors etc. This component is positioned above the bottom tab.
  * Need to get the height of bottom tab via `useBottomTabBarHeight()` hook to be called on screen.
  * */
-export function OceanInterface (): JSX.Element {
-  const { height, err: e } = useSelector((state: RootState) => state.ocean)
-  const transaction = useSelector((state: RootState) => firstTransactionSelector(state.ocean))
-  const [tx, setTx] = useState<OceanTransaction>(transaction)
+export function OceanInterface (): JSX.Element | null {
   const dispatch = useDispatch()
   const client = useWhaleApiClient()
+
+  // store
+  const { height, err: e } = useSelector((state: RootState) => state.ocean)
+  const transaction = useSelector((state: RootState) => firstTransactionSelector(state.ocean))
+
+  // state
+  const [tx, setTx] = useState<OceanTransaction|undefined>(transaction)
   const [err, setError] = useState<Error | undefined>(e)
+
   const slideAnim = useRef(new Animated.Value(0)).current
   Animated.timing(slideAnim, { toValue: height, duration: 300, useNativeDriver: false }).start()
 
+  const dismissDrawer = useCallback(() => {
+    setTx(undefined)
+    setError(undefined)
+  }, [])
+
   useEffect(() => {
+    // last available job will remained in this UI state until get dismissed
     if (transaction !== undefined) {
-      broadcastTransaction(transaction, client, setError).then(() => {
-        setTx({
-          ...transaction,
-          title: translate('screens/OceanInterface', 'Transaction complete'),
-          broadcasted: true
-        })
-        dispatch(ocean.actions.popTransaction())
-      }).catch(() => {
+      setTx({
+        ...transaction,
+        title: translate('screens/OceanInterface', 'Transaction complete'),
+        broadcasted: false
       })
+
+      broadcastTransaction(transaction, client)
+        .catch((e: Error) => setError(e))
+        .finally(() => dispatch(ocean.actions.popTransaction())) // remove the job as soon as completion
     }
   }, [transaction])
 
+  if (tx === undefined) {
+    return null
+  }
+
   return (
     <Animated.View
       style={[tailwind('bg-white px-5 py-3 flex-row absolute w-full items-center border-t border-gray-200 z-10'), {
         bottom: slideAnim,
         height: 75
       }]}
     >
-      {err !== undefined ? <TransactionError txid={tx?.signed.txId} /> : <TransactionDetail tx={tx} />}
+      {
+        err !== undefined
+          ? <TransactionError txid={tx.signed.txId} onClose={dismissDrawer} />
+          : <TransactionDetail tx={tx} onClose={dismissDrawer} />
+      }
     </Animated.View>
   )
 }
 
-function TransactionDetail ({ tx }: { tx: OceanTransaction }): JSX.Element {
+function TransactionDetail ({ tx, onClose }: { tx: OceanTransaction, onClose: () => void }): JSX.Element {
```

Comment:
remove `dispatch` pass on to all sub-components in this page, isolate the logic and renderer